pkg/proxy/proxy.go

```diff
@@ -200,8 +204,20 @@ func (s *Proxy) handleWebsocket(w http.ResponseWriter, req *http.Request) {
 		failed(err)
 		return
 	}
+	dnsNames := []string{u.Hostname()}
+	ips := []string{u.Hostname()}
+	if client.Spec.ExternalKubeAPIEnabled && len(client.Spec.Connection.ExternalKubernetesAPIEndpoint) > 0 {
+		eu, err := url.Parse(client.Spec.Connection.ExternalKubernetesAPIEndpoint)
+		if err != nil {
+			klog.Errorf("Failed to get external host %#v", err)
```

Comment:
Do we need a return here?

Reply:
no, if external ip failed to allocate, origin tower ability to access the member cluster is still working, So just go ahead, when the external IP available, svcWatcher will update the cluster resource.

Reply:
OK, thanks for the explanation